### PR TITLE
Added private events

### PR DESCRIPTION
**Pull request by Nydauron**
Added a field `IsPrivate` to the Event model that is only accessible to Staff and above. Anyone else that searches for an event will not see this field. Any events that have `IsPrivate: true` will also be hidden from the user.

### CODE REVIEW

**Comment by Timothy-Gonzalez** [comment]
Looks like this is blocked by #501, so I'll review this once that can be merged.

**Comment by Timothy-Gonzalez** [comment]
#501 just merged, so this needs to be updated to fix the conflicts. Have fun!

**Review by Timothy-Gonzalez — CHANGES_REQUESTED** [review]
#501 just merged, so this needs to be updated to fix the conflicts. Have fun!

**Review by asehgal220 — APPROVED** [review]

**Comment by asehgal220** [comment]
From what I can tell, other than the merge conflicts everything looks fine.

**Comment by Timothy-Gonzalez** [comment]
#520 has merge conflicts with this, other than that it looks good.

One small thing might be switching to use common config for token secret, which will be added in #518. We can discuss this also, but I thing is makes sense to have it in common config since it is going to end up being the same `TOKEN_SECRET` in every single config otherwise.

File: services/event/config/config.go
Comment (Timothy-Gonzalez) [inline services/event/config/config.go:72]: Since `TOKEN_SECRET` exists in common now, we shouldn't need to define it like this for every file, and instead just use common_config.

File: services/event/service/token_service.go
Comment (Timothy-Gonzalez) [inline services/event/service/token_service.go:15]: This seems like a method that might want to be common, which would also make using the token secret much easier.

I don't see anything really event specific here.

File: services/event/tests/event_test.go
Comment (Timothy-Gonzalez) [inline services/event/tests/event_test.go:1044]: Small thing, might be a bigger project to do in a different PR (if so, create an issue for this), it is at the same scale as reformatting everything, but if we could convert all these magic "testid1" to constant variables that would be amazing.

File: tests/e2e/events_test/get_filtered_events_test.go
Comment (Timothy-Gonzalez) [inline tests/e2e/events_test/get_filtered_events_test.go:706]: I feel like this testevent1 creation can be extracted out into a function instead of being duplicated 50 times.

**Review by Timothy-Gonzalez — CHANGES_REQUESTED** [review]

File: services/event/config/config.go
Comment (Nydauron) [reply services/event/config/config.go]: Nice catch. Yeah i set those up prior to making the decision to put `TOKEN_SECRET` in `common`.

File: services/event/tests/event_test.go
Comment (Nydauron) [reply services/event/tests/event_test.go:1044]: Yeah, this is something that should be resolved in a separate PR. I can go ahead and make an issue for that.

File: tests/e2e/events_test/get_filtered_events_test.go
Comment (Nydauron) [reply tests/e2e/events_test/get_filtered_events_test.go:706]: This can be also addressed in the issue that I'd say I'd make.

**Review by Timothy-Gonzalez — APPROVED** [review]

**Review by asehgal220 — APPROVED** [review]
Great Job!

**Pull request merged**
